src/libs/AxelarQueryAPI.ts

```diff
@@ -229,73 +242,142 @@ export class AxelarQueryAPI {
       return {
         baseFee,
         expressFee,
-        sourceToken: source_token,
+        sourceToken: source_token as BaseFeeResponse["sourceToken"],
         executeGasMultiplier: parseFloat(execute_gas_multiplier.toFixed(2)),
         destToken: {
           gas_price: destination_native_token.gas_price,
-          gas_price_gwei: parseInt(destination_native_token.gas_price_gwei).toString(),
           decimals: destination_native_token.decimals,
+          token_price: destination_native_token.token_price,
+          name: destination_native_token.name,
+          symbol: destination_native_token.symbol,
+          l1_gas_price_in_units: destination_native_token.l1_gas_price_in_units,
         },
+        ethereumToken: ethereum_token as BaseFeeResponse["ethereumToken"],
         apiResponse: response,
         success: true,
         expressSupported: express_supported,
       };
     });
   }
 
+  public async estimateL1GasFee(destChainId: EvmChain | string, l1FeeParams: EstimateL1FeeParams) {
+    // Retrieve the RPC URL for the source chain to calculate L1 fee
+    const rpcMap = this.environment === "mainnet" ? mainnetRpcMap : testnetRpcMap;
+
+    // Throw an error if the RPC URL is not found
+    if (!rpcMap[destChainId]) {
+      throw new Error(`RPC URL not found for chain ${destChainId}`);
+    }
+
+    const provider = new ethers.providers.JsonRpcProvider(rpcMap[destChainId]);
+
+    return getL1FeeForL2(provider, destChainId, l1FeeParams);
+  }
+
+  public async calculateL1FeeForDestL2(
+    destChainId: EvmChain | string,
+    destToken: FeeToken,
+    executeData: `0x${string}` | undefined,
+    sourceToken: FeeToken,
+    ethereumToken: BaseFeeResponse["ethereumToken"],
+    actualGasMultiplier: number
+  ): Promise<[BigNumber, BigNumber]> {
+    let l1ExecutionFee = BigNumber.from(0);
+    let l1ExecutionFeeWithMultiplier = BigNumber.from(0);
+
+    // If the destination chain is L2, calculate the L1 execution fee
+    const isDestinationChainL2 = await isL2Chain(this.environment, destChainId, destToken);
+
+    if (isDestinationChainL2) {
+      if (!executeData) {
+        console.warn(
+          `Since you did not provide executeData, this API will not accurately calculate the
+          total required fee as we will not be able to capture the L1 inclusion fee for this L2 chain.`
+        );
+      }
+
+      if (!destToken.l1_gas_price_in_units) {
```

Comment:
just realized, this `l1_gas_price_in_units` field may not be here on all L2 destination chains. [check arbitrum, for example]. so with this logic, this error will always throw for arbitrum